src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/RcaConf.java

```diff
@@ -186,21 +193,29 @@ public DeciderConfig getDeciderConfig() {
     return ImmutableMap.copyOf(conf.getRcaConfigSettings());
   }
 
+  public <T> T readRcaConfig(String rcaName, String key, T defaultValue, Class<? extends T> clazz) {
+    return readRcaConfig(rcaName, key, defaultValue, (s) -> true, clazz);
+  }
+
   @SuppressWarnings("unchecked")
-  public <T> T readRcaConfig(String rcaName, String key, Class<? extends T> clazz) {
+  public <T> T readRcaConfig(String rcaName, String key, T defaultValue, Predicate<T> validator, Class<? extends T> clazz) {
```

Comment:
Did you completely remove `public <T> T readRcaConfig(String rcaName, String key, Class<? extends T> clazz)` ? No more consumers for it?

Reply:
I have not removed the method without predicate. I have refactored consumers using this method. It does not have any more consumers.

Reply:
What are you thoughts on removing that method if it does not have any consumers?